tests/rpm-ostree/main.yml

```diff
@@ -0,0 +1,565 @@
+---
+# vim: set ft=ansible:
+#
+#   This is a general test suite for rpm-ostree that exercises several of the
+# rpm-ostree subcommands.  These tests assume the atomic host is provisioned
+# with the latest version.  These tests are meant to be run in succession.
+#
+- name: rpm-ostree - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  roles:
+    # This playbook requires Ansible 2.2 and an Atomic Host
+    - role: ansible_version_check
+      avc_major: "2"
+      avc_minor: "2"
+      tags:
+        - ansible_version_check
+
+    # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+#####################################################################################
+#
+# rpm-ostree deploy version
+#   - validates deploying by version string
+#   - validates cleanup of rollback deployment
+#
+#####################################################################################
+
+- name: rpm-ostree - deploy by version
+  hosts: all
+  become: yes
+
+  tags:
+    - deploy_version
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - include: roles/rpm_ostree_status/tasks/main.yml
+
+    - name: Set current version and refspec
+      set_fact:
+         head_version: "{{ ros_booted['version'] }}"
+         refspec: "{{ ros_booted['origin'] }}"
+
+    - name: Pull last two commits
+      command: ostree pull --commit-metadata-only --depth=1 {{ refspec }}
+      register: osp
+      until: osp.rc == 0
+      retries: 5
+      delay: 5
+
+    - name: Get HEAD-1 version
+      shell: ostree show $(ostree rev-parse {{ refspec }}^) --print-metadata-key version | tr -d \'
+      register: hmo_version
+
+    - name: Deploy version {{ hmo_version.stdout }}
```

Comment:
Seems like it does although I have seen instances when it doesn't (possibly in loops?)

```
TASK [Deploy version 7.2017.383] ***********************************************
changed: [fedora] => {
    "attempts": 1, 
    "changed": true, 
    "cmd": [
        "rpm-ostree", 
        "deploy", 
        "26.55"
    ], 
    "delta": "0:00:32.515279", 
    "end": "2017-06-06 14:46:03.710649", 
    "rc": 0, 
    "start": "2017-06-06 14:45:31.195370", 
    "warnings": []
}
```